Source/WebCore/rendering/svg/LegacyRenderSVGForeignObject.cpp

```diff
@@ -54,8 +56,16 @@ SVGForeignObjectElement& LegacyRenderSVGForeignObject::foreignObjectElement() co
     return downcast<SVGForeignObjectElement>(RenderSVGBlock::graphicsElement());
 }
 
+bool LegacyRenderSVGForeignObject::isUnderSVGRoot() const
+{
+    return (nullptr != ancestorsOfType<LegacyRenderSVGRoot>(*this).first());
+}
+
 void LegacyRenderSVGForeignObject::paint(PaintInfo& paintInfo, const LayoutPoint&)
 {
+    if (!isUnderSVGRoot())
```

Comment:
can we just use `SVGRenderSupport::findTreeRootObject()` instead of introducing a new method that does the same thing?